pydash/pydash_app/user/services/pruning.py

```diff
@@ -0,0 +1,36 @@
+"""
+Provides functionality to periodically remove all users that have not verified their account.
+"""
+
+from datetime import timedelta
+
+import periodic_tasks
+import pydash_app.user.repository
+import pydash_logger
+
+logger = pydash_logger.Logger(__name__)
+
+
+def schedule_periodic_pruning_task(
+        interval=timedelta(days=1),
+        scheduler=periodic_tasks.default_task_scheduler):
+    scheduler.add_periodic_task(
+        name=('users', 'pruning'),
+        task=_prune_unverified_users,
+        interval=interval,
+        scheduler=scheduler)
+
+
+def _prune_unverified_users():
+    users_to_remove = []
+
+    for user in pydash_app.user.repository.all():
```

Comment:
Instead of iterating over all users, I believe we may be able to iterate only over the set of unverified users, since we remove them from that set once they have verified.
_I'm currently working out how to implement this._